src/lib/components/card/index.tsx

```diff
@@ -1,10 +1,14 @@
-import { ReactNode } from 'react';
+import { ElementType, ReactNode } from 'react';
 import classNamesUtil from 'classnames';
 import { ChevronRightIcon } from '../icon';
 
 import styles from './style.module.scss';
 
-export interface CardProps {
+const CardDefault = 'section' as const
+type CardDefaultAsType = typeof CardDefault;
+
+type CardOwnProps<E extends ElementType = CardDefaultAsType> = {
+  as?: E;
```

Comment:
Not necessarily a blocker from merging the PR but, would we want to narrow the type of elements we could render the `Card` component as — i.e, buttons, anchor, section, etc — or do we want to keep it wide? 🤔

Reply:
That's a very good point! My main concern about this is on how to do it on a way that doesn't block react components like `react-router` or `next` Link components, WDYT?

Reply:
That makes sense. Let's leave it as it is :)